File: et-web/src/main/webapp/assets/js/services/user-service.js
/**
 * Created by VMakarenko on 2/7/15.
 */

angular.module("app")
    .service("UserService", ['$http',
        function UserService($http) {
            var url = "api/user";

            this.checkUsername = function (username) {
                return $http.get(url + "/checkUsername?username=" + username );
            };

            this.createUser = function (user) {
                return $http.post(url, user );
            };


        }
    ]);

File: et-ejb/src/main/java/ru/vmakarenko/services/UserService.java
package ru.vmakarenko.services;

import ru.vmakarenko.common.LoginPassCheckResult;
import ru.vmakarenko.entities.User;

/**
 * Created by VMakarenko on 2/4/15.
 */
public interface UserService {
    public User getByPrincipal(String user);

    void createUser(User user);

    LoginPassCheckResult validateUsername(String username);

    LoginPassCheckResult vaidateUser(User user);
}
